tests/system_tests_tcp_adaptor_tls.py

```diff
@@ -610,3 +613,114 @@ def test_connector_bad_server_cert(self):
+                            'privateKeyFile': cls.ssl_file('server-private-key.pem'),
+                            'password': 'server-password'}),
+            ('listener', {'role': 'inter-router', 'port': inter_router_port})
+        ])
+
+        cls.connector_name = 'connectorToBeDeleted'
+        cls.connector_props = {
+            'port': cls.nginx_port,
+            'address': 'examples',
+            'host': 'localhost',
+            'name': cls.connector_name,
+            # Verifies host name. The host name in the certificate sent by the server must match 'localhost'
+            'verifyHostname': 'yes',
+            'sslProfile': 'http-connector-ssl-profile'
+        }
+        config_qdrb = Qdrouterd.Config([
+            ('router', {'mode': 'interior', 'id': 'QDR.B'}),
+            ('listener', {'port': cls.tester.get_port(), 'role': 'normal', 'host': '0.0.0.0'}),
+            ('tcpConnector', cls.connector_props),
+            ('connector', {'name': 'connectorToA', 'role': 'inter-router',
+                           'port': inter_router_port,
+                           'verifyHostname': 'yes'}),
+            ('sslProfile', {'name': 'http-connector-ssl-profile',
+                            'caCertFile': cls.ssl_file('ca-certificate.pem'),
+                            'certFile': cls.ssl_file('client-certificate.pem'),
+                            'privateKeyFile': cls.ssl_file('client-private-key.pem'),
+                            'password': 'client-password'}),
+        ])
+        cls.router_qdra = cls.tester.qdrouterd("tcp-two-router-tls-A", config_qdra, wait=True)
+        cls.router_qdrb = cls.tester.qdrouterd("tcp-two-router-tls-B", config_qdrb)
+        cls.router_qdra.wait_router_connected('QDR.B')
+        cls.router_qdrb.wait_router_connected('QDR.A')
+        wait_tcp_listeners_up(cls.router_qdra.addresses[0])
+
+        # curl will use these additional args to connect to the router.
+        cls.curl_args = ['--cacert', cls.ssl_file('ca-certificate.pem'), '--cert-type', 'PEM',
+                         '--cert', cls.ssl_file('client-certificate.pem') + ":client-password",
+                         '--key', cls.ssl_file('client-private-key.pem')]
+
+    def test_get_image_jpg_http2(self):
+        image_file_name = '/test.jpg'
+        address = get_address(self.router_qdra) + "/images" + image_file_name
+        out = run_curl(args=[address, '--verbose'] + self.curl_args +
+                            ['--output', self.router_qdra.outdir + image_file_name] +
+                            ['--http2', '--http2-prior-knowledge'])
+        digest_of_server_file = get_digest(image_file(image_file(image_file_name[1:])))
+        digest_of_response_file = get_digest(self.router_qdra.outdir + image_file_name)
+        self.assertEqual(digest_of_server_file, digest_of_response_file)
+
+    def test_get_image_jpg_http11(self):
+        image_file_name = '/test.jpg'
```

Comment:
Is the trex image free to use without citing source? ;P

Reply:
Yes, I got it from a free to use copyright free website